srcs/builtin/builtin_exit.c

```diff
@@ -6,20 +6,76 @@
 /*   By: jjaniec <jjaniec@student.42.fr>            +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2018/04/25 17:21:10 by sbrucker          #+#    #+#             */
-/*   Updated: 2018/08/09 17:11:40 by jjaniec          ###   ########.fr       */
+/*   Updated: 2018/09/16 19:24:02 by jjaniec          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include <twenty_one_sh.h>
 
-inline void	builtin_exit(char **argv, char **envp, t_exec *exe)
-{
-	(void)argv;
-	(void)envp;
+/*
+** Free shell malloced data before exiting
+*/
 
+static void	exit_free_sh_data(t_exec *exe)
+{
 	if (exe->tmp_envp)
 		ft_free_argv(exe->tmp_envp);
 	if (exe->envp)
 		ft_free_argv(exe->envp);
-	exit(0);
+}
+
+/*
+** Check if a string is only composed of numbers
+*/
+
+static int	str_is_num(char *str)
+{
+	if (!(str && *str))
+		return (1);
+	while (str && *str && ft_isdigit(*str))
+		str++;
+	if (!(*str))
+		return (1);
+	return (0);
+}
+
+/*
+** Exit shell and free malloced data,
+** if an argument is specified and this argument is numerical,
+** return argument value as exit value,
+** if the passed argument is non numerical, exit w/ 256,
+** otherwise exit with 0 by default
+*/
+
+inline void	builtin_exit(char **argv, char **envp, t_exec *exe)
+{
+	int		exit_val;
+	int		first_arg_is_num;
+
+	(void)envp;
+	exit_val = 0;
+	first_arg_is_num = 0;
+	if (argv && argv[1])
+	{
+		if ((first_arg_is_num = str_is_num(argv[1])))
+			exit_val = ft_atoi(argv[1]);
+		if (!first_arg_is_num || argv[2])
+		{
```

Comment:
This whole if is useless too, because you go like this
```
if (A || B)
{
    if (A) { do something }
    else if (B) { do something }
} else { do something }
```
But you can do it like this :
```
if (A) { do something }
else if (B) { do something }
else { do something }
```
It will be the same !

Reply:
And more readable :)